cpp/src/arrow/memory_pool.cc

```diff
@@ -88,22 +88,43 @@ MemoryPool::~MemoryPool() {}
 
 int64_t MemoryPool::max_memory() const { return -1; }
```

Comment:
Remove `include <mutex>` in this file for good measure?

Reply:
Oh, right.